compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -869,10 +869,15 @@ checkIface :: MonadGamma m => Module -> DefInterface -> m ()
 checkIface m iface = do
   
   -- check view
-  tycon <- match _TCon (EExpectedViewType viewtype) viewtype
+  let (func, _) = viewtype ^. _TypeSplitApp
+  tycon <- match _TCon (EExpectedViewType "non-type-constructor" viewtype) func
```

Comment:
Perhaps it would be nicer here to explicitly match on `func` and give `EExpectedViewType` a different string for each non-`TCon` case

Reply:
done